src/Scope.ts

```diff
@@ -204,7 +232,7 @@ export class Scope {
     * @param namespaceName - the current namespace name
     */
     public hasInterface(ifaceName: string, namespaceName?: string): boolean {
-        return !!this.hasInterface(ifaceName, namespaceName);
+        return !!this.getInterface(ifaceName, namespaceName);
```

Comment:
This was a infinite loop mistake that slipped by!

Reply:
Yikes! nice catch.